rdm/render.py

```diff
@@ -34,6 +34,34 @@ def join_to(foreign_keys, table, primary_key='id'):
     return joined
 
 
+def md_indent(snippet, header_shift=0):
+    lines = snippet.split('\n')
+    i = 0
+    in_code = False
+    if header_shift < 0:
+        while i > header_shift:
+            i -= 1
+            for j in range(0, len(lines)):
+                if lines[j][0:3] == "```":
+                    in_code = not in_code
+                if in_code:
+                    continue
+                if lines[j][0] == '#':
+                    lines[j] = lines[j][1:].lstrip()
+    else:
+        while i < header_shift:
+            i += 1
+            for j in range(0, len(lines)):
```

Comment:
I think this would be slightly cleaner using the `enumerate` function:

```
for j, line in enumerate(lines):
```

BTW there are some goodies here: https://docs.python.org/3/library/functions.html

You probably have seen most of them, but it's worth spending some time browsing through if you haven't.